src/main/webapp/script.js

```diff
@@ -55,3 +56,122 @@ window.addEventListener("load", function(){
+  }
+  else {
+    slideshowImage.className = "carousel-item";
+  }
+  slideshowImage.appendChild(createImgElement(blobkey));
+  var caption = document.createElement("div");
+  caption.className = "carousel-caption";
+  var header = document.createElement("H3");
+  var name = document.createTextNode(name); 
+  header.appendChild(name);
+  caption.appendChild(header);
+  var para = document.createElement("P");
+  var bio = document.createTextNode(bio);
+  para.appendChild(bio);
+  caption.appendChild(para);
+  slideshowImage.appendChild(caption); 
+  return slideshowImage;
+}
+
+function addIndicators(numPhotos) {
+    let indicators = document.getElementById('carousel-indicators');
+    for (var i = 0; i < numPhotos; i++) {
+      var singleIndicator = document.createElement("LI");
+      singleIndicator.setAttribute("data-target", "#feed");
+      singleIndicator.setAttribute("data-slide-to", i);
+      if (i === 0) {
+        singleIndicator.className = "active";
+      }
+      indicators.appendChild(singleIndicator);
+    }
+}
+
+function getNextPotentialMatch() {
+  deletePotentialMatchInfo();
+  let currentUser = getCurrentUserId();
+  fetch('/potential-matches?userid=' + currentUser).then(response => response.text()).then((pmID) => { 
+      let noMatch = "NO_POTENTIAL_MATCHES";
+      if (pmID === noMatch) {
+        noPotentialMatch();
+        return;
+      }
+      document.getElementById("pass-btn").disabled = false;
+      document.getElementById("friend-btn").disabled = false;
+      displayPotentialMatchInfo(pmID);
+  }); 
+}
+
+function noPotentialMatch() {
+  document.getElementById("pass-btn").disabled = true;
+  document.getElementById("friend-btn").disabled = true;
+  let noMatchImg = createSlideshowElement("images/nomatches.png", true, "", "");
+  let carouselContainer = document.getElementById("carousel-inner");
+  carouselContainer.appendChild(noMatchImg);
+}
+
+function matchButtonPressed() {
+  getNextPotentialMatch();  
+}
+
+function passButtonPressed() {
+  getNextPotentialMatch();
+}
+
+function loadProfile() {
+  let id = getCurrentUserId();
+  if (id === null) {
+    return;
+  }
+  console.log(id);
+  fetch('/user-data?id=' + id).then(response => response.json()).then((userinfo) => {
+    console.log(userinfo);
+    name = userinfo.name;
+    bio = userinfo.bio;
+    document.getElementById("name").value = name;
+    document.getElementById("bio").value = bio;
+  });
+}
+
+function changeImgPath(blobKey, id) {
+  fetch('/blob-key?imageKey='+blobKey).then((response) => {
+    console.log(response);
+    return response.blob();
+  }).then((blobContent) => {
+    var blobURL = URL.createObjectURL(blobContent);
+    document.getElementById(id).src = blobURL;
+  });  
+}
+
+function createImgElement(imageURL) {
+  const imgElement = document.createElement('img');
+  imgElement.src = imageURL;
+  imgElement.setAttribute("height", "800");
+  imgElement.setAttribute("width", "1100");
```

Comment:
Should these be scaled based on the image aspect ratio?

Alternatively, you can provide either a `width` or `height` alone to cause the other dimension to be scaled automatically.

Reply:
I kinda wanted both the height and the width to be fixed so that the slideshow images all had the same dimensions, do you think that having it scaled automatically would be better?

Reply:
No that sounds like a good reason!